authentication/passwordless.go

```diff
@@ -92,12 +105,27 @@ func (p *Passwordless) LoginWithSMS(ctx context.Context, params passwordless.Log
 	return
 }
 
-func (p *Passwordless) addClientAuthentication(params *oauth.ClientAuthentication) {
+func (p *Passwordless) addClientAuthentication(params *oauth.ClientAuthentication) error {
 	if params.ClientID == "" {
 		params.ClientID = p.authentication.clientID
 	}
 
-	if params.ClientSecret == "" && p.authentication.clientSecret != "" {
+	if p.authentication.clientAssertionSigningKey != "" {
```

Comment:
Should we also check that `clientAssertionSigningAlg` is not empty?

Reply:
Fixed in 4f1ea30418bb3789e920307da5e65652520252f1 (also changed in oauth.go)